include/dali/core/math_util.h

```diff
@@ -192,6 +192,46 @@ float sinc(float x) {
   return std::sin(x) / x;
 }
 
+/// @brief Calculates power for integer arguments
+template <typename X, typename Y>
+DALI_HOST_DEV DALI_FORCEINLINE
+std::enable_if_t<std::is_integral<X>::value && std::is_unsigned<Y>::value,
+                  decltype(std::declval<X>() * std::declval<Y>())>
```

Comment:
I don't think this is a good promotion - perhaps we should just keep the result of type `X` or do something smarter with the signedness. I think that if X is signed and Y is unsigned, the result would be unsigned and that's not what you'd expect:
`ipow(-3, 3u)` should be -27, not 4294967269
```suggestion
                  decltype(std::declval<X>() * std::declval<std::make_signed_t<Y>>())>
```

Reply:
How about the type of X{} *  X{} as I suggested in other comment?

Reply:
Sounds reasonable.